### PR TITLE
DateTime

### PR DESCRIPTION
**Pull request by isc30**
Implemented a subset of `DateTime` in `mscorlib`

- [x] New Unit Test for DateTime
- [x] [Constructors](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,189)
- [x] [Static Equals](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,730)
- [x] [DayOfWeek](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,1011)
- [x] [Now](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,1105)
- [x] [Today](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,013b44cc4074ffd2)
- [x] [AddMilliseconds](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,6ab0bc8b0fc7eb33)
- [x] [AddSeconds](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,4a8bd6c0b5c72d81)
- [x] [AddMinutes](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,569965820bb5ae3d)
- [x] [AddHours](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,1673702782a58887)
- [x] [AddDays](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,f5cb6edd98581ca2)
- [x] [ToLongDateString](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,b35520d1ad2aa159)
- [x] [ToLongTimeString](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,ae03e97e178476d2)
- [x] [ToShortDateString](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,43ddbaa5fb33ab4e)
- [x] [ToShortTimeString](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,e63c8cfb6c10c534)
- [x] [ToString](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,11d38858dedad166)
- [x] [ToString(string format)](https://source.dot.net/#System.Private.CoreLib/shared/System/DateTime.cs,5088b43558351160)

image

### CODE REVIEW

File: src/DSharp.Compiler/Importer/MetadataImporter.cs [hunk 1]
```diff
@@ -428,7 +428,7 @@ private void ImportMethods(TypeSymbol typeSymbol)
             {
                 if (method.IsSpecialName)
                 {
-                    continue;
+                    //continue;
```
Comment (fred-perkins) [inline src/DSharp.Compiler/Importer/MetadataImporter.cs]: Comment here, Do we want to remove the continue?

File: src/DSharp.Compiler/Importer/MetadataImporter.cs [hunk 2]
```diff
@@ -489,6 +489,7 @@ private void ImportMethods(TypeSymbol typeSymbol)
 
                 if (string.IsNullOrEmpty(transformedName) == false)
                 {
+                    Console.WriteLine("0: " + transformedName);
```
Comment (fred-perkins) [inline src/DSharp.Compiler/Importer/MetadataImporter.cs]: Do we want to keep console write lines? I don't care too much honestly

File: src/DSharp.Compiler/Importer/MetadataImporter.cs [hunk 3]
```diff
@@ -576,20 +577,28 @@ private void ImportProperties(TypeSymbol typeSymbol)
 
                         if (string.IsNullOrEmpty(transformedName) == false)
                         {
-                            fieldSymbol.SetTransformName(transformedName);
+                            Console.WriteLine("1: " + transformedName);
```
Comment (fred-perkins) [inline src/DSharp.Compiler/Importer/MetadataImporter.cs]: Same as other console comment :)

File: src/DSharp.Compiler/ScriptModel/Symbols/NullTypeSymbol.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿namespace DSharp.Compiler.ScriptModel.Symbols
+{
+    internal class NullTypeSymbol : TypeSymbol
+    {
+        public NullTypeSymbol()
+            : base(SymbolType.Namespace, "", new NamespaceSymbol("", new SymbolSet()))
```
Comment (fred-perkins) [inline src/DSharp.Compiler/ScriptModel/Symbols/NullTypeSymbol.cs]: Can we use `String.Empty` instead of `""`

**Comment by isc30** [comment]
closed because of pending changes

**Comment by isc30** [comment]
👏

**Pull request merged**